modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheContext.java

```diff
@@ -2350,6 +2352,29 @@ public boolean hasContinuousQueryListeners(@Nullable IgniteInternalTx tx) {
             contQryMgr.notifyContinuousQueries(tx) && !F.isEmpty(contQryMgr.updateListeners(false, false));
     }
 
+    /**
+     * Apply changes from {@link SchemaAddQueryEntityOperation}
+     *
+     * @param op Add query entity schema operation.
+     */
+    public void onSchemaAddQueryEntity(SchemaAddQueryEntityOperation op) {
+        CacheConfiguration oldCfg = cacheCfg;
```

Comment:
Let's use 
```
        CacheConfiguration<?, ?> oldCfg = cacheCfg;

        if (oldCfg != null) {
            CacheConfiguration<?, ?> newCfg = new CacheConfiguration<>(oldCfg);
```

to remove compiler warnings.

Reply:
Can this method be invoked concurrently?

Reply:
Yes, i copy reference on stack for thread safety. We should ensure that we write to cacheCfg reference to fully constructed object.